odoo/addons/base/tests/test_user_has_group.py

```diff
@@ -249,3 +248,41 @@ def test_implied_groups(self):
 
         with self.assertRaises(ValidationError):
             user_b.write({"groups_id": [(4, group_C.id)]})
+
+    def test_has_group_cleared_cache_on_write(self):
+        Users = self.env["res.users"]
+        self.env.registry._clear_cache()
+        self.assertEqual(
+            len(Users.pool._Registry__cache),
+            0,
+            "Ensure cache is empty (got: %r)" % self.env.cache._data['res.users']
+        )
+
+        def populate_cache():
+            self.test_user.has_group('test_user_has_group.group0')
+            self.assertEqual(
+                len(Users.pool._Registry__cache),
+                1,
+                "has_group chache method must be populated"
+            )
```

Comment:
All you need is checking that the cache is nonempty.
```suggestion
            self.assertTrue(self.registry._Registry__cache, "user.has_group cache must be populated")
```